### PR TITLE
[move-prover] Rewrite the access control spec as two state invariants

### PR DESCRIPTION
**Pull request by junkil-park**
- This PR defines `is_txn_signer`, and replaced `is_signer` with it because the previous implementation of `is_signer` didn't fit for purpose (see the issue https://github.com/diem/diem/issues/9018.).

- The existing access control spec uses the schema application which is verbose. To simplify the access control spec and make it easier to read, this PR rewrites the existing access control spec as two state invariants with the new spec construct `is_txn_signer_addr` (and `is_txn_signer`).

- This PR is mostly about cleaning-up the spec for the role-based access control in the Diem Framework. The next step is to look into the capability-based one (e.g., MintCapability, ...)

Resolves https://github.com/diem/diem/issues/9018.

## Motivation

To rewrite the access control spec with two-state invariants

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

cargo test

### CODE REVIEW

File: language/diem-framework/modules/DiemVMConfig.move [hunk 1]
```diff
@@ -218,6 +218,18 @@ module DiemFramework::DiemVMConfig {
 
     /// # Access Control
 
+    /// The permission "UpdateVMConfig" is granted to DiemRoot [[H11]][PERMISSION].
+    spec module {
+        invariant [global, isolated] forall addr: address where exists<DiemConfig<DiemVMConfig>>(addr):
+            addr == @DiemRoot;
+
+        invariant update old(DiemConfig::spec_is_published<DiemVMConfig>())
```
Comment (DavidLDill) [inline language/diem-framework/modules/DiemVMConfig.move:226]: I noticed that some specs condition the conclusion whether a resource is published both in old state and current states (as in here), and some just check the old state.   Is this a historical accident, or do we really need to condition on exists in both old and current state sometimes?

**Review by DavidLDill — APPROVED** [review]

File: language/diem-framework/modules/DiemVMConfig.move (hunk 1 above)
Comment (junkil-park) [reply language/diem-framework/modules/DiemVMConfig.move:226]: That's a good point! This is related to the issue of partial functions in spec. To be on a safe side, I think we should check the existence of a certain resource on both the old state and the current state. In some cases (like this case), it's ok to check it on the old state only when there is a "resource persistence" invariant in place, so the existence on the current state is implied.

**Pull request closed without merging**